### ISSUE TITLE
Ability to login with device code for Microsoft account more than once

### ISSUE THREAD

**Issue by UnknownCoding** [PrismarineJS/mineflayer#2233]
## Is your feature request related to a problem? Please describe.
I haven't found a way to login to more than one account using the device auth (which is the most ideal way as it doesn't require me to get 2fa code every time I want the account to login

## Describe the solution you'd like
ability to set username in options and it find the correct stuff corelated with logging into the account externally/where ever it is currently stored

## Describe alternatives you've considered
maybe have it store the login file in the current folder to allow for each folder to be a different account, idk just thought it might help

## Additional context
no additional content

**Comment by extremeheat** [comment]
Do not supply a password if you want to use device code auth. Multiple accounts can be used as usual by using a different username per account.

**Comment by UnknownCoding** [comment]
> Do not supply a password if you want to use device code auth. Multiple accounts can be used as usual by using a different username per account.

im not using a password, this is the createbot as listed below, when i run it even if its in a new folder it just states that the account is already connected to the server instead of letting me onto said server
```
const bot = mineflayer.createBot({
  host: 'redacted', // minecraft server ip
  /*
  host: 'redacted', // minecraft test server ip
  port: 25554,                // only set if you need a port that isn't 25565 */
  // username: 'email@example.com', // minecraft username
  // password: '12345678' // minecraft password, comment out if you want to log into online-mode=false servers
  // version: false,             // only set if you need a specific version or snapshot (ie: "1.8.9" or "1.16.5"), otherwise it's set automatically
  auth: 'microsoft'              // only set if you need microsoft auth, then set this to 'microsoft'
})
```

**Comment by extremeheat** [comment]
Yes that’s because the default username is empty string. You need to put something there. As defined in the docs, the username picks which account to use.

**Comment by UnknownCoding** [comment]
> Yes that’s because the default username is empty string. You need to put something there. As defined in the docs, the username picks which account to use.

It should be labeled in the example as well as it's not super clear when the example has an email address there and when searching for Microsoft in documentation I was unable to find it

**Comment by extremeheat** [comment]
Yes you can put your email there also. There's no example anywhere I know that tells you to comment the username out like you did. You aren't supposed to. In the Minecraft protocol docs it explains that the field is interpreted differently behind the scenes for device code auth, but you should still specify it no matter the auth type.
